components/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/token/AbstractJWTGenerator.java

```diff
@@ -0,0 +1,552 @@
+            = new ConcurrentHashMap<String, Integer>();
+
+    //constructor for testing purposes
+    public AbstractJWTGenerator(boolean includeClaims, boolean enableSigning) {
+        this.includeClaims = includeClaims;
+        this.enableSigning = enableSigning;
+        signatureAlgorithm = NONE;
+    }
+
+    /**
+     * Reads the ClaimsRetrieverImplClass from app-manager.xml ->
+     * AppConsumerAuthConfiguration -> ClaimsRetrieverImplClass.
+     *
+     * @throws org.wso2.carbon.appmgt.api.AppManagementException
+     */
+    public AbstractJWTGenerator() {
+
+        String claimsRetrieverImplClass = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
+                getAPIManagerConfiguration().getFirstProperty(ClaimsRetriever.CLAIMS_RETRIEVER_IMPL_CLASS);
+        dialectURI = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
+                getAPIManagerConfiguration().getFirstProperty(ClaimsRetriever.CONSUMER_DIALECT_URI);
+        if (dialectURI == null) {
+            dialectURI = ClaimsRetriever.DEFAULT_DIALECT_URI;
+        }
+        if (claimsRetrieverImplClass != null) {
+            try {
+                claimsRetriever = (ClaimsRetriever) Class.forName(claimsRetrieverImplClass).newInstance();
+                claimsRetriever.init();
+            } catch (ClassNotFoundException e) {
+                log.error("Cannot find class: " + claimsRetrieverImplClass, e);
+            } catch (InstantiationException e) {
+                log.error("Error instantiating " + claimsRetrieverImplClass);
+            } catch (IllegalAccessException e) {
+                log.error("Illegal access to " + claimsRetrieverImplClass);
+            } catch (AppManagementException e) {
+                log.error("Error while initializing " + claimsRetrieverImplClass);
+            }
+        }
+
+        signatureAlgorithm = ServiceReferenceHolder.getInstance().getAPIManagerConfigurationService().
+                getAPIManagerConfiguration().getFirstProperty(AppMConstants.SIGNATURE_ALGORITHM);
+        if (signatureAlgorithm == null || !(signatureAlgorithm.equals(NONE) || signatureAlgorithm.equals(
+                SHA256_WITH_RSA))) {
+            signatureAlgorithm = SHA256_WITH_RSA;
+        }
+
+        addClaimsSelectively = Boolean.parseBoolean(ServiceReferenceHolder.getInstance().
+                getAPIManagerConfigurationService().getAPIManagerConfiguration().
+                getFirstProperty(AppMConstants.API_CONSUMER_AUTHENTICATION_ADD_CLAIMS_SELECTIVELY));
+    }
+
+    /**
+     * Method that generates the JWT token from SAML2 response
+     *
+     * @param saml2Assertions
+     * @param webAppInfoDTO
+     * @param messageContext
+     * @return jwt token
+     * @throws org.wso2.carbon.appmgt.api.AppManagementException
+     */
+    public String generateToken(Map<String, Object> saml2Assertions, WebAppInfoDTO webAppInfoDTO,
```

Comment:
See the possibility of passing an org.wso2.carbon.appmgt.api.model.WebApp instead of WebAppInfoDTO